src/Test.Utilities/CodeFixTestBase.cs

```diff
@@ -167,51 +209,26 @@ private void VerifyFixAll(string language, DiagnosticAnalyzer analyzerOpt, CodeF
         }
 
-        private void VerifyFixAll(
+        private Document ApplyFixAll(
             Document document,
-            DiagnosticAnalyzer analyzerOpt,
+            CodeAction codeAction,
             CodeFixProvider codeFixProvider,
-            string newSource,
-            string newSourceFileName,
-            IEnumerable<TestAdditionalDocument> additionalFiles,
-            bool allowNewCompilerDiagnostics,
-            TestValidationMode validationMode)
+            FixAllProvider fixAllProvider,
+            string diagnosticIdToFix,
+            IEnumerable<Diagnostic> fixableDiagnostics)
         {
-            var fixableDiagnosticIds = codeFixProvider.FixableDiagnosticIds.ToSet();
-            Func<IEnumerable<Diagnostic>, ImmutableArray<Diagnostic>> getFixableDiagnostics = diags =>
-                diags.Where(d => fixableDiagnosticIds.Contains(d.Id)).ToImmutableArrayOrEmpty();
-
-            var analyzerDiagnostics = GetSortedDiagnostics(analyzerOpt, new[] { document }, additionalFiles: additionalFiles, validationMode: validationMode);
-            var compilerDiagnostics = document.GetSemanticModelAsync().Result.GetDiagnostics();
-            var fixableDiagnostics = getFixableDiagnostics(analyzerDiagnostics.Concat(compilerDiagnostics));
-
-            var fixAllProvider = codeFixProvider.GetFixAllProvider();
-            var diagnosticProvider = new FixAllDiagnosticProvider(analyzerOpt, additionalFiles, validationMode, getFixableDiagnostics);
-            var fixAllContext = new FixAllContext(document, codeFixProvider, FixAllScope.Document, string.Empty, fixableDiagnostics.Select(d => d.Id), diagnosticProvider, CancellationToken.None);
-            var codeAction = fixAllProvider.GetFixAsync(fixAllContext).Result;
-            document = document.Apply(codeAction);
-            additionalFiles = document.Project.AdditionalDocuments.Select(a => new TestAdditionalDocument(a));
-
-            additionalFiles = document.Project.AdditionalDocuments.Select(a => new TestAdditionalDocument(a));
-
-            analyzerDiagnostics = GetSortedDiagnostics(analyzerOpt, new[] { document }, additionalFiles: additionalFiles, validationMode: validationMode);
-
-            var updatedCompilerDiagnostics = document.GetSemanticModelAsync().Result.GetDiagnostics();
-            var newCompilerDiagnostics = GetNewDiagnostics(compilerDiagnostics, updatedCompilerDiagnostics);
-            if (!allowNewCompilerDiagnostics && newCompilerDiagnostics.Any())
+            var diagnosticProvider = new FixAllDiagnosticProvider(fixableDiagnostics);
+            IEnumerable<string> fixableDiagnosticIds = new string[] { diagnosticIdToFix };
+            var fixAllContext = new FixAllContext(document, codeFixProvider, FixAllScope.Document, codeAction.EquivalenceKey, fixableDiagnosticIds, diagnosticProvider, CancellationToken.None);
+            var fixTask = fixAllProvider.GetFixAsync(fixAllContext);
```

Comment:
how this can return null?

Reply:
Ah, I added that while debugging the nullref due to dotnet/roslyn#23492. Let me revert.